src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.PackTool.targets

```diff
@@ -10,20 +10,91 @@ Copyright (c) .NET Foundation. All rights reserved.
 ***********************************************************************************************
 -->
 
+<!--
+
+NOTE: This file is imported from the following contexts, so be aware when writing Conditions and thinking about evaluation order:
+* cross-targeting (multi-TFM) builds.
+  * these may not have a RID or RIDs set, but still need to be able to pack tools
+  * IsCrossTargetingBuild is set to true in this case
+* 'outer' builds for a single TFM
+  * these builds are RID-agnostic, but still need to be able to pack 'inner', RID-specific tools.
+  * IsCrossTargetingBuild is unset in this case, RuntimeIdentifier is unset, and RuntimeIdentifiers or ToolPackageRuntimeIdentifiers is set to a list of RIDs
+  * if these builds are RID-specific but not AOT, then we pack the outer tool without implementation dlls and the N inner-RID-specific tools
+  * if these builds are RID-specific and AOT, then we pack the outer tool only without implementation dlls
+  * if these builds are RID-agnostic, then we pack the outer tool only with implementation dlls
+* 'inner' builds for a single TFM
+  * these builds are RID-specific, and have RuntimeIdentifier set to a single RID, and RuntimeIdentifiers or ToolPackageRuntimeIdentifiers set to a list of RIDs
+  * these builds pack only the inner, RID-specific tool implementation dlls and have a unique PackageType of DotnetToolRidPackage
+-->
+
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.GenerateToolsSettingsFile"
             AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.GenerateShims"
             AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
-  <UsingTask TaskName="Microsoft.NET.Build.Tasks.ResolveToolPackagePaths"
-            AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.GetEmbeddedApphostPaths"
       AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.GetNuGetShortFolderName"
              AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
-  <UsingTask TaskName="Microsoft.NET.Build.Tasks.RemoveTargetFromList"
-            AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
+  <PropertyGroup>
+    <!-- tools are specially-formatted packages, so we tell nuget Pack to not even try to include build output -->
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+
+    <!-- the publish* properties _can_ be set, but only for the 'inner' RID-specific builds. We need to make sure that for the outer, agnostic build they are unset  -->
+    <PublishSelfContained Condition="'$(RuntimeIdentifier)' == ''">false</PublishSelfContained>
```

Comment:
What about `PublishRuntimeIdentifier` 😏  I think it's ok since this is called after RID inference targets. Same with this one, tbh.

Reply:
I didn't realize PublishRuntimeIdentifier was a thing! You're right though that ordering takes care of this one for us, but we should be consistent and comprehensive if we can. I'll add it.